### DISCUSSION TITLE
ideas

### DISCUSSION THREAD

**Discussion by dotbot123** [landgreen/n-gon#371] [Ideas]
some kind of tech that makes bullets affected by pilot wave

also some ideas from a while ago (don't know if these have been added, this was like 2 years ago)

tech idea: removes your field but adds a shield that has 50 health and regens +1 shield every second, set your defense to 0 and you can never gain more.
tech branch: +10 max shield, -25% damage

power-up idea: gain +25% of max energy
tech for it: like quenching but for energy power-ups

weapon or field: vortex, pulls mobs and blocks towards you (if weapon, wherever you shoot it) like wormhole does with blocks and drains energy and damages them for about laser damage
tech for it: +100% pull power, -10% dps
also tech for it: (ONLY FOR WEAPON) charge, you can charge your vortex for up to 5 secs and +10% damage
and pull per sec

field: Phase, you can phase through vertical-facing walls for huge energy drain
tech for it: phasing creates a permanent tunnel through walls, +100% energy drain

tech: infinity theory/theorem, if you fire while out of ammo, +25% of original ammo, but it's corroded (-25% damage from corroded ammo)

JUNK tech: infinite damage, but only 15 bullets per level

JUNK tech: you always move toward your mouse (left, right, or up, like the spaceship junk) and deal damage to mobs that hit you

JUNK tech: convert all mobs to bosses, but they only drop 5-10 health or ammo (or energy power-up)

JUNK tech: remove all blocks (excluding elevators and switches) until the final boss

JUNK tech: lock testing mode again.

tech: give a permanent +100% damage, +50 max health and energy, but +10% chance to get code.blank (in tech menu you have a chance to get all blank upgrades, regardless of how many slots you have, and you cannot research code.blank) basically, it cancels your tech immediately

weapon: blade, slash a short distance in front of you, hits multiple mobs and deals 2x laser damage, has infinite ammo, and does not drain energy

a new level: it is dark and only lit up a short distance away, like cloaking, mobs can only see you in the light.

tech: set fire rate to 0.1 and double your ammo, but -50% damage

**Comment by AnnonymousNerd87** [comment]
thanks for moving this to discussions

**Reply by dotbot123** [reply]
np :)

**Comment by dotbot123** [comment]
entanglement: entangle 2 blocks/mobs, when 1 moves or dies, the other copies it
(somewhat similar to domino power from dishonored 2)

**Comment by dotbot123** [comment]
ooh i want a tech that allows me to reenter a wormhole (maybe makes it so mobs can travel through them too?)

**Reply by whymycodenowork** [reply]
there is one for projectiles

**Comment by AnnonymousNerd87** [comment]
or, alternatively, it's an upgrade for dark energy, mobs within are
entangled (using a special font tag) but take 1/2 damage, not MACHO

**Reply by dotbot123** [reply]
that's cool

**Comment by AnnonymousNerd87** [comment]
that's called the wormhole field

**Reply by dotbot123** [reply]
dude, it's a tech FOR wormhole field

**Reply by dotbot123** [reply]
reread it

**Comment by AnnonymousNerd87** [comment]
ahhh, like geodesics

**Comment by AnnonymousNerd87** [comment]
the problem is it could trap you in a loop if you have low velocity

**Reply by dotbot123** [reply]
yeah, just add a cooldown

**Comment by whymycodenowork** [comment]
junk idea: backtrack
send you to the previous level
defeated enemies and collected powerups will not respawn
if you are on initial then it send you to final or just kills you idk
or it could swap the entrance and exit doors

**Reply by dotbot123** [reply]
good thought

**Comment by dotbot123** [comment]
doppler shift: something with hue shift as different speeds
probably junk tech

**Reply by landgreen** [reply]
I'll add that to my todo list.

**Reply by dotbot123** [reply]
yay

**Comment by cubingpenguin** [comment]
Good ideas

**Comment by cubingpenguin** [comment]
JUNKtech idea: Peto's Paradox
Larger entities are less vulnerable to certain types of damage [I was thinking radiation, explosions, lasers], and smaller entities [including M] are more vulnerable.

Also I have an idea for a field that lets you use your field to change size or maybe ghost

**Reply by cubingpenguin** [reply]
Also, just a non-tech idea I feel like throwing in:
what if constraints like "no harm reduction" or "no damage buffs" were added

**Reply by cubingpenguin** [reply]
plus, just wondering, what does the white bar above my health mean?

**Reply by kgurchiek** [reply]
@cubingpenguin that indicates your current harm reduction stat

**Reply by dotbot123** [reply]
for the last part, timestop has a tech that does that partly, and wormhole has tech for that

**Reply by cubingpenguin** [reply]
not the size change right?

**Reply by dotbot123** [reply]
not the size change

**Reply by dotbot123** [reply]
the size change is cool

**Comment by dotbot123** [comment]
switches laser to ammo for a boost

**Reply by btcool** [reply]
Maybe the ammo is a battery? I always thought that laser should make ammo give energy, probably about 3 second's of fire cost.

**Comment by dotbot123** [comment]
combines plasma torch/extruder/plasma bot with laser and boosts damage, reduces energy cost, makes the range shorter?

**Comment by ohwthestupid** [comment]
when you say energy power up do you mean the coupling?

**Reply by dotbot123** [reply]
no, i meant a new power up that is like a health power up but for energy

**Comment by dotbot123** [comment]
if you reach 5k ammo, your ammo becomes infinite

**Comment by aeiou-13579** [comment]
mine techs that make direct hits more useful or easy, like
more direct hit damage,
detonated mine deals extra damage (or some other effect) if the mine previously dealt a direct hit,
spawn a secondary (possibly mini) mine if mine deals direct hit,
make mines not arc [while standing?] (like rocket-propelled grenade)

m i n e   r i c o c h e t

**Reply by aeiou-13579** [reply]
mines detonate on hit

**Comment by aeiou-13579** [comment]
foam grenade

**Comment by AnnonymousNerd87** [comment]
missiles that start slow but accelerate until hypersonic

**Reply by landgreen** [reply]
This is a good idea, although I can't let missiles go too fast or else they penetrate walls.

**Reply by dotbot123** [reply]
![Screenshot 2025-05-06 7 31 56 AM](https://github.com/user-attachments/assets/125c2e89-14fb-407a-af1e-5e6fdfb02a46)

**Comment by AnnonymousNerd87** [comment]
yeah, I was thinking more like nailgun speed than actually hypersonic, you
might also increase the turning speed to prevent it from being too weak

**Comment by AnnonymousNerd87** [comment]
hmm, adding it to another tech could be cool, like maybe ICBMs that
accelerate so they can actually have decent range without lifetime
upgrades, or launch system/(as I call it) micromissiles that swarm at high
speed.

**Comment by whymycodenowork** [comment]
maybe a tech that makes the pilot wave size adjustable or the pilot wave works through walls or the ability to just grab blocks and fling them using the mouse kinda like the mouse from  animation vs animator

**Reply by dotbot123** [reply]
yes

**Reply by cubingpenguin** [reply]
Maybe it should be like:
Pilot wave does not diminish while out of line of sight
Tech have -N choices

N should be ≤ 3.

**Comment by cubingpenguin** [comment]
JUNK idea from a friend: engineer
+50% JUNK chance
Spawn 2 tech. One has all possible normal choices and one has all possible JUNK choices.

**Reply by aeiou-13579** [reply]
you'd just be able to have 0% junk tech using the junk tech defragment

**Comment by dotbot123** [comment]
please add back boss health bars

**Reply by aeiou-13579** [reply]
just shoot them :)

**Comment by TechnoGuru123698741** [comment]
First i want to thank the creator of this beautiful game for his hard work! I really love playing it in my free time.
Next the suggestion:
Unified field theory
when paused you can click to change your field
2x frequency for field

Problem:
disability to switch fields fast

Suggestions:
ability to change fields using keyboard shortcut and mouse wheel
my variants:

    You can change fields by pressing shift+number(1/2/3...0) or by pressing shift+q/e or by shift+mouse scroll
    You can change fields when crouching and pressing number(1/2/3...0) or when crouching and pressing q/e or by crouching + mouse scroll
    When paused pressing numbers/ q or e/ mouse scroll changes field instead of guns

**Comment by btcool** [comment]
Idea: Shotgun Slug
Guntech for shotgun
Fires one large shotgun slug with 5x damage instead of multiple pellets.
Mutually exclusive with: needle gun, nail-shot, foam-shot, ice-shot, siphonaptera, nematodes
Either exclusive to or requires Rivot gun
I see no reason to use an image other than [Wikipedia's](https://en.wikipedia.org/wiki/Shotgun_slug) [image](https://en.wikipedia.org/wiki/File:Brenneke_Slug-1.jpg).

**Reply by aeiou-13579** [reply]
I feel like rivet gun is already enough

**Reply by cubingpenguin** [reply]
What if there was a branch that made it fire one giant worm?

**Comment by aeiou-13579** [comment]
tech that removes junk%, because only the junk tech defragment does that and it's not 'captured' by coherence

1. Commercial cleaning | use __ research, set junk chance to 0% (1 research per 3%)
2. Contamination control | clicking cancel for..., -3% chance for junk choices
3. Abrasive blasting | 1.1X damage, removing this decreases junk chance by -10%
4. Restoration | +6% initial junk chance, at the start of each level -2% junk chance
5. Parts cleaning | for each gun you have, decrease junk chance by -2%

**Reply by btcool** [reply]
I love these. Here's an implementation of the first one:

```
{

	name: "commercial cleaning",
	descriptionFunction() {
		return `use ${powerUps.orb.research(tech.junkChance*100)} to remove ${tech.junkChance*100}% junk chance`
	},
        isInstant: true,
	maxCount: 1,
	count: 0,
	frequency: 2,
	frequencyDefault: 2,
	allowed() {
		return (tech.junkChance > 0.05) && (tech.junkChance*100 <= powerUps.research.count) 
	},
	requires: "More research than junk chance, junk chance at least 5%",
	effect() {
		powerUps.research.count -= 100*tech.junkChance;
                tech.removeJunkTechFromPool(tech.junkChance);
	},
	remove() {
	}
}
```
Having it be 1 to 1 with research and junk might not be fair, so you might want to change it to 1 research to 3 junk. 

The third one could be
```
{
	name: "abrasive blasting",
	descriptionFunction() {
		return `<strong>1.1x</strong> <strong class='color-d'>damage</strong><br><span class='color-remove'>removing</span> this decreases <strong class='color-junk'>JUNK</strong> by <strong>10%</strong>`
	},
	maxCount: 1,
	count: 0,
	frequency: 2,
	frequencyDefault: 2,
        isRemoveBenefit: true,
	allowed() {
		return true
	},
	requires: "",
	effect() {
		m.damageDone*=1.1;
	},
	remove() {
               tech.removeJunkTechFromPool(0.1);
               m.damageDone/=1.1;
	}
}
```

**Reply by aeiou-13579** [reply]
names are definitely going to be cleaning related
https://en.wikipedia.org/wiki/Cleaning

I think these fit well:
1. Commercial cleaning
2. Contamination control
3. Abrasive blasting (removing material with very fast sand)
4. Restoration
5. Parts cleaning

some other cool/thematic names:
Ultrasonic cleaning - shake particles loose using ultrasound
Hygiene
Flame cleaning
Cleaning agent
Cleaning symbiosis
Sanitation
Sterilization
Self-cleaning
(long name) [Ultraviolet] germicidal irradiation

a lot of these have the word 'cleaning'

**Reply by btcool** [reply]
I figured out what the fourth one means(thanks annonyomousnerd87 for the explanation) but I can't think of how i'd write it.
 
I had an idea how to write the second:
```
{
	name: "contamination control",
	descriptionFunction() {
		return `clicking <strong class='color-cancel'>cancel</strong> decreases <strong class='color-junk'>JUNK</strong> by <strong>3%</strong>`
	},
	maxCount: 1,
	count: 0,
	frequency: 2,
	frequencyDefault: 2,
	allowed() {
		return true
	},
	requires: "",
	effect() {
		tech.isCancelJunk=true
	},
	remove() {
                tech.isCancelJunk=false
	}
}

```
In powerup.js, in the endDraft function


> if (tech.isCancelCouple) powerUps.spawnDelay("coupling", 10)
**if (tech.isCancelJunk) {
        tech.removeJunkFromPool(0.03);
}**
if (tech.isCancelTech && tech.cancelTechCount === 0 && type !== "entanglement") {
                tech.cancelTechCount++
                // powerUps.research.use('tech')
                powerUps[type].effect();
                return
            }

@landgreen, if you like the code I wrote, feel free to use it. I'd love for a note in that patch thanking aeiou-13579 and btcool, but I can't exactly sue you if you don't.

**Reply by aeiou-13579** [reply]
with these, I think you should be able to have negative junk chance,
which might not do anything but prevent future increases to junk chance

**Reply by aeiou-13579** [reply]
or there could be techs that synergize with negative junk chance

**Reply by cubingpenguin** [reply]
Or maybe even have "anti-junk" that only spawns with negative JUNK chance

**Comment by aeiou-13579** [comment]
also make interest and coherence have some junk chance because they are insta picks with no drawback but heuristics has +5% for no good reason

**Reply by landgreen** [reply]
Coherence and interest are allowed to be good because they have tech you must already have before they are activated, and because they don't give a benefit for a few levels.  On the highest difficulties or late in the game they are not insta picks.

Coherence requires you to already have decoherence which already gives you some control over tech options.
Interest only shines once you have a coupling tech, or you have saved up research for a few levels, which isn't easy on the harder difficulties.

**Reply by aeiou-13579** [reply]
i see

**Comment by AnnonymousNerd87** [comment]
rivet gun is not enough, big slugs sounds awesome.

**Comment by AnnonymousNerd87** [comment]
it starts at +8% and it goes down by 2% each level, until it starts
removing junk

**Comment by AnnonymousNerd87** [comment]
Environmental Initiative (or something else idc): Whenever you remove JUNK%
10% chance to spawn a random powerup per JUNK% removed

**Reply by aeiou-13579** [reply]
yess, that's a good one

**Reply by btcool** [reply]
```
{
	name: "enviromental initiative",
	descriptionFunction() {
		return `removing  <strong class='color-junk'>JUNK</strong> spawns random powerups`
	},
	maxCount: 1,
	count: 0,
	frequency: 2,
	frequencyDefault: 2,
	allowed() {
		return (tech.junkChance >= 0.1)
	},
	requires: "at least 10% junk",
	effect() {
		tech.isJunkSpawnPowerUps=true
	},
	remove() {
                tech.isJunkSpawnPowerUps=false
	}
}
```
In tech.js, in the removeJunkTechFromPool function:
>tech.junkChance -= percent
if (tech.junkChance < 0.001 || tech.junkChance === undefined) tech.junkChance = 0
if (tech.junkChance > 1) tech.junkChance = 1
**const types = ["heal", "ammo", "heal", "ammo", "research", "coupling", "boost", "tech", "gun", "field"]
for (let i = 0; I < Math.floor(percent*100); i++) powerUps.spawnDelay(types[Math.floor(Math.random() * types.length)], 1)**

Imma add the techs to my mod when I get the chance

**Reply by landgreen** [reply]
Sorry ya'll, I'm not planning to add a tech to remove JUNK beyond the current JUNK tech that does that already.

<img width="342" alt="Screenshot 2025-05-16 at 7 35 46 AM" src="https://github.com/user-attachments/assets/11b36fd5-3728-4f87-8993-f6a7ab740e32" />

**Reply by dotbot123** [reply]
![is-there-a-name-for-the-disintegrating-emoji-v0-91z9dlqeat5c1](https://github.com/user-attachments/assets/f5e48c69-bc47-4edd-a62f-ce4500ffa5d8)

**Comment by AnnonymousNerd87** [comment]
yeah I wasn't taking it seriously, seems like something from an overhaul
mod but I'm not planing on another of those after I accidentally merged
dumb n-gon with n-gon and I can't revert merge pull requests.

**Comment by dotbot123** [comment]
let pilot wave move grenades+mines+sporeganium

**Reply by landgreen** [reply]
mines and sporeganium get attached to the walls, but I could make pilot wave grab them before the hit a wall.  same with any bullets

**Reply by dotbot123** [reply]
can you please add that as a tech?

**Reply by landgreen** [reply]
I was thinking it would be better as a baseline change to pilot wave.  I would apply it to all bullets, so it would nerf nailgun, superballs which scales with speed, but buff mines and spores since you could deliver them

**Reply by dotbot123** [reply]
alr, thanks

**Reply by landgreen** [reply]
ok I got the pilot wave extended to bullets.  I made it ignore fast moving bullets, since it doesn't have an advantage.  The best interactions are mines, laser mines, and spores/worms.  Drones, waves are fun, but not useful.

**Reply by cubingpenguin** [reply]
Interesting. Should lead to some great new strats! "Bells Theorem" would be a little problematic, though.

**Comment by AnnonymousNerd87** [comment]
seem interesting

**Comment by AnnonymousNerd87** [comment]
I'm imagining it creates an event horizon which swallows up even photons
from laser and freezes time in the area and you can stack like 20 seconds
of laser damage on a single point. Def. not happening, but it would be
funny.

**Reply by dotbot123** [reply]
the laser does not create projectiles, so yea not going to happen

**Comment by AnnonymousNerd87** [comment]
I know

**Comment by AnnonymousNerd87** [comment]
actually it does I checked the code but it wouldn't work because they're
weird
